### PR TITLE
first draft string_indices

### PR DESCRIPTION
**Pull request by phoniks**

### CODE REVIEW

File: spec_3/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+let example = "Catch me outside, how about that?"
+
+const dropEmpty = (array) => {
+  console.log(array)
```
Comment (jrobcodes) [inline spec_3/index.js]: Remove `console.log`

File: spec_3/index.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  let newArray = []
+  for(i = 0; i < array.length; i++){
+    let currentWord = array[i]
+    if( currentWord){
+      newArray.push(currentWord)
+    }
+  }
+  console.log('new',newArray)
```
Comment (jrobcodes) [inline spec_3/index.js]: Remove `console.log`

File: spec_3/index.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+let example = "Catch me outside, how about that?"
+
+const dropEmpty = (array) => {
+  console.log(array)
+  let newArray = []
+  for(i = 0; i < array.length; i++){
```
Comment (jrobcodes) [inline spec_3/index.js]: When I see the pattern: variable collection declaration, for loop, return declared variable, this stands out to me as an opportunity for more concise code using functional methods on the collection object:
```js
const dropEmpty = array =>
  array.filter( word => word !== undefined )
```

File: spec_3/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  return phraseAsArray[index - 1]
+}
+
+const splitOnSpace = (word) => {
+  const wordArray = word.split(' ')
+  const cleanedArray = wordArray.map(stripPunctuation)
+  return cleanedArray
+}
+
+const stripPunctuation = (word) => {
```
Comment (jrobcodes) [inline spec_3/index.js]: Single parameter functions may omit the parentheses in ES6.  Single statement functions, where the statement is a `return`, may omit the `return` keyword and braces.

File: spec_3/index.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+const stringIndex = (index, phrase) => {
+  let phraseAsArray = splitOnSpace(phrase)
+  console.log(phraseAsArray)
+  phraseAsArray = dropEmpty(phraseAsArray)
+  return phraseAsArray[index - 1]
+}
+
+const splitOnSpace = (word) => {
+  const wordArray = word.split(' ')
```
Comment (jrobcodes) [inline spec_3/index.js]: This function again presents the opportunity for a more concise version by chaining the operations:
```js
const splitOnSpace = word =>
  word.split( ' ' ).map( stripPunctuation )
```

File: spec_3/index.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      newArray.push(currentWord)
+    }
+  }
+  console.log('new',newArray)
+  return newArray
+}
+
+const stringIndex = (index, phrase) => {
+  let phraseAsArray = splitOnSpace(phrase)
+  console.log(phraseAsArray)
```
Comment (jrobcodes) [inline spec_3/index.js]: Remove `console.log`

File: spec_3/index.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    let currentWord = array[i]
+    if( currentWord){
+      newArray.push(currentWord)
+    }
+  }
+  console.log('new',newArray)
+  return newArray
+}
+
+const stringIndex = (index, phrase) => {
```
Comment (jrobcodes) [inline spec_3/index.js]: I *think* this function could be more clearly named `stringAtIndex`?

**Review by jrobcodes — COMMENTED** [review]
There are not tests for this code.

File: spec_3/index.js (hunk 4 above)
Comment (phoniks) [reply spec_3/index.js]: people mention this to me a lot, and while I'm aware of it, I'm not sure what the advantage of omitting them is beside saving a single keystroke.  can you enlighten me? Is there a performance benefit or is this just a style choice?

Comment (jrobcodes) [reply spec_3/index.js]: Consistency in style.  Note that I just pointed it out, I didn't recommend a change.  😄   For me, it is much less visual noise when parsing simple functions.

**Pull request merged**